lib/c-api/src/wasm_c_api/externals/function.rs

```diff
@@ -154,18 +154,34 @@ pub unsafe extern "C" fn wasm_func_call(
                 .into_iter()
                 .map(TryInto::try_into)
                 .collect::<Result<Vec<Val>, _>>()
-                .expect("Argument conversion failed")
+                .expect("Arguments conversion failed")
         })
         .unwrap_or_default();
 
     match func.inner.call(&params) {
         Ok(wasm_results) => {
-            *results = wasm_results
+            let vals = wasm_results
                 .into_iter()
                 .map(TryInto::try_into)
                 .collect::<Result<Vec<wasm_val_t>, _>>()
-                .expect("Argument conversion failed")
-                .into();
+                .expect("Results conversion failed");
+
+            // `results` is an uninitialized vector. Set a new value.
+            if results.size == 0 || results.data.is_null() {
```

Comment:
Should we implement a function like `is_uninitialized` function for `wasm_$name_vec_t`? I believe yes.

Reply:
Done.

Reply:
I think having such a function would make our intentions clear in the code.